save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/LnkUserProjectController.kt

```diff
@@ -46,14 +46,52 @@ class LnkUserProjectController(
             ?: throw NoSuchElementException("There is no $projectName project in $organizationName organization")
         val usersWithRoles = if (projectPermissionEvaluator.hasPermission(authentication, project, Permission.READ)) {
             lnkUserProjectService.getAllUsersAndRolesByProject(project)
-                .filter { (_, role) -> role != Role.NONE }
+                .filter { (_, role) ->
+                    role != Role.NONE
+                }
                 .map { (user, role) ->
                     user.toUserInfo(mapOf(project.organization.name + "/" + project.name to role))
                 }
-                .also { logger.trace("Found ${it.size} users for project: $it") }
+                .also {
+                    logger.trace("Found ${it.size} users for project: $it")
+                }
         } else {
             emptyList()
         }
         return usersWithRoles
     }
+
+    /**
+     * @param organizationName
+     * @param projectName
+     * @param authentication
+     * @param prefix
+     * @return list of users, not connected to the project
+     * @throws NoSuchElementException
+     */
+    @GetMapping("/users/not-from/{organizationName}/{projectName}")
+    @Suppress("UnsafeCallOnNullableType")
+    fun getAllUsersNotFromProjectWithNamesStartingWith(
```

Comment:
Yes, ok, done

Reply:
I think it should be a 404 error instead of empty list. Because the user requests all users not from a project, that they cannot see (have no read permission), and hence it shouldn't look like the request finishes correctly. The same as you are now throwing exception if the project doesn't exist.

Reply:
Now it might be better.

Reply:
OK, good